File: src/main/java/me/m64diamondstar/ingeniamccore/database/MySQL.java
package me.m64diamondstar.ingeniamccore.database;

import me.m64diamondstar.ingeniamccore.Main;
import org.bukkit.Bukkit;

import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.SQLException;

public class MySQL {

    final private String host;
    final private String port;
    final private String database;
    final private String username;
    final private String password;

    private Connection connection;

    public MySQL(Main main){

        port = main.getConfig().getString("MySQL.Port");
        database = main.getConfig().getString("MySQL.Database");
        username = main.getConfig().getString("MySQL.Username");
        password = main.getConfig().getString("MySQL.Password");
        host = main.getConfig().getString("MySQL.Host");

        try {
            this.connect();
        } catch (SQLException e) {
            Bukkit.getLogger().info("Database not connected ✗");
        }

        if(this.isConnected()){
            Bukkit.getLogger().info("Database is connected ✓");
        }


    }

    public boolean isConnected(){
        return (connection != null);
    }

    public void connect() throws SQLException {
        if(isConnected()) return;
        connection = DriverManager.getConnection("jdbc:mysql://" + username + ":" + password + "@" + host + ":" + port + "/" + database + "?useSSL=false",
                username, password);
    }

    public void disconnect(){
        if(!isConnected()) return;
        try{
            connection.close();
        }catch (SQLException throwables) {
            throwables.printStackTrace();
        }
    }

    public Connection getConnection(){
        return this.connection;
    }

}

File: src/main/java/me/m64diamondstar/ingeniamccore/database/MySQL.kt
package me.m64diamondstar.ingeniamccore.database

import me.m64diamondstar.ingeniamccore.Main
import org.bukkit.Bukkit
import java.sql.*
import kotlin.Throws

class MySQL(main: Main) {
    private val host: String
    private val port: String
    private val database: String
    private val username: String
    private val password: String
    var connection: Connection? = null
        private set
    val isConnected: Boolean
        get() = connection != null

    @Throws(SQLException::class)
    fun connect() {
        if (isConnected) return
        connection = DriverManager.getConnection(
            "jdbc:mysql://$username:$password@$host:$port/$database?useSSL=false",
            username, password
        )
    }

    fun disconnect() {
        if (!isConnected) return
        try {
            connection!!.close()
        } catch (throwables: SQLException) {
            throwables.printStackTrace()
        }
    }

    init {
        port = main.config.getString("MySQL.Port")!!
        database = main.config.getString("MySQL.Database")!!
        username = main.config.getString("MySQL.Username")!!
        password = main.config.getString("MySQL.Password")!!
        host = main.config.getString("MySQL.Host")!!
        try {
            connect()
        } catch (e: SQLException) {
            Bukkit.getLogger().info("Database not connected ✗")
        }
        if (isConnected) {
            Bukkit.getLogger().info("Database is connected ✓")
        }
    }
}